agent/src/bpf/audit.bpf.c

```diff
@@ -212,12 +268,30 @@ record_blob_data (struct pt_regs *ctx)
   event->base.type = AUDIT_DATA_BLOB;
   err = bpf_probe_read_user_str (event->base.key, KEY_SIZE, (void *)key_ptr);
   if (err < 0)
-    goto error;
+    {
+      DEBUG ("unable to read event key: %ld\n", err);
+      goto error;
+    }
 
-  value_size &= (VALUE_SIZE - 1);
-  err = bpf_probe_read_user (event->value, value_size, (void *)value_ptr);
-  if (err < 0)
-    goto error;
+  if (value_size > 0)
+    {
+      long value_ptr;
+
+      err = bpf_usdt_arg (ctx, 2, &value_ptr);
+      if (err < 0)
+	{
+	  DEBUG ("unable to get the third argument: %ld\n", err);
+	  goto error;
+	}
+
+      value_size &= (VALUE_SIZE - 1);
```

Comment:
I am not sure about how this works. Is the goal to ignore some upper bits? And how likely is it that `VALUE_SIZE` will someday become not equal to 2^n and this blows?

Reply:
Without this, the eBPF verifier complains: "R2 min value is negative, either use unsigned or 'var &= const'". I suspect there is no other way to tell the verifier than using this idiom.